### DISCUSSION TITLE
Think about making mastercontainer read-only

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#3319] [Ideas] [answered]
Follow-up to https://github.com/nextcloud/all-in-one/issues/2506

- [ ] needs https://github.com/nextcloud/all-in-one/pull/3069

TODO:
- [ ] Create anonymous volumes in Dockerfile that are automatically created for all directories that need to be writeable
- [ ] https://github.com/nextcloud/all-in-one/pull/3215
- [ ] put www-data into root group by default so that we dont need to add it to a group during runtimes
- [ ] adjust docs to reflect read-only setting
- [ ] more?

**Comment by szaimen** [comment]
> * put www-data into root group by default so that we dont need to add it to a group during runtimes? Does it come with security problems?

@Zoey2936 do you know if this comes with security draw-backs?

**Comment by Zoey2936** [comment]
> > * put www-data into root group by default so that we dont need to add it to a group during runtimes? Does it come with security problems?
> 
> @Zoey2936 do you know if this comes with security draw-backs?

not sure, but it could cause that the www-data user can access all files

**Comment by szaimen** [comment]
I see but would that be a security problem? I mean it runs in a dedicated and soon read-only container. The most important files in /mnt/docker-aio-config are already readable by the user anyway... So do you think it would be fine to do this?

**Comment by Zoey2936** [comment]
also not sure

**Comment by szaimen** [comment]
I asked if the idea makes sense internally and it makes sense to do this 👍

**Comment by szaimen** [comment, marked as answer]
I've changed my mind. I decided to not do this for now as it will also make testing (much) harder which is of course not cool from a maintenance perspective.
